packages/components/inputs/localized-rich-text-input/src/localized-rich-text-input.tsx

```diff
@@ -181,160 +195,181 @@ const expandedTranslationsReducer = (
+      );
+    }
+
+    const initialExpandedTranslationsState = Object.keys(props.value).reduce(
+      (translations, locale) => {
+        return {
+          [locale]: Boolean(props.defaultExpandMultilineText),
+          ...translations,
+        };
+      },
+      {}
     );
-  }
 
-  if (props.showExpandIcon) {
-    warning(
-      typeof props.onClickExpand === 'function',
-      'LocalizedRichTextInput: "onClickExpand" is required when showExpandIcon is true'
+    const [expandedTranslationsState, expandedTranslationsDispatch] =
+      useReducer(expandedTranslationsReducer, initialExpandedTranslationsState);
+
+    const defaultExpansionState = Boolean(
+      props.hideLanguageExpansionControls || props.defaultExpandLanguages
     );
-  }
 
-  if (props.hideLanguageExpansionControls) {
-    warning(
-      typeof props.defaultExpandLanguages !== 'boolean',
-      'LocalizedRichTextInput: "defaultExpandLanguages" does not have any effect when "hideLanguageExpansionControls" is set.'
+    const [areLanguagesOpened, toggleLanguages] = useToggleState(
+      defaultExpansionState
     );
-  }
 
-  const initialExpandedTranslationsState = Object.keys(props.value).reduce(
-    (translations, locale) => {
-      return {
-        [locale]: Boolean(props.defaultExpandMultilineText),
-        ...translations,
-      };
-    },
-    {}
-  );
-
-  const [expandedTranslationsState, expandedTranslationsDispatch] = useReducer(
-    expandedTranslationsReducer,
-    initialExpandedTranslationsState
-  );
-
-  const defaultExpansionState =
-    props.hideLanguageExpansionControls ||
-    props.defaultExpandLanguages ||
-    // useToggleState's default is `true`, but we want `false`
-    false;
-
-  const [areLanguagesOpened, toggleLanguages] = useToggleState(
-    defaultExpansionState
-  );
-
-  const toggleLanguage = useCallback(
-    (language) => {
-      expandedTranslationsDispatch({ type: 'toggle', payload: language });
-    },
-    [expandedTranslationsDispatch]
-  );
-
-  const languages = sortLanguages(
-    props.selectedLanguage,
-    Object.keys(props.value)
-  );
-
-  const hasErrorInRemainingLanguages =
-    props.hasError ||
-    getHasErrorOnRemainingLanguages(props.errors, props.selectedLanguage);
-
-  const hasWarningInRemainingLanguages =
-    props.hasWarning ||
-    getHasWarningOnRemainingLanguages(props.warnings, props.selectedLanguage);
-
-  if (hasErrorInRemainingLanguages || hasWarningInRemainingLanguages) {
-    if (!areLanguagesOpened) {
-      // this update within render replaces the old `getDerivedStateFromProps` functionality
-      // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
-      toggleLanguages();
+    const toggleLanguage = useCallback(
+      (language) => {
+        expandedTranslationsDispatch({ type: 'toggle', payload: language });
+      },
+      [expandedTranslationsDispatch]
+    );
+
+    const handleChange = useCallback(
```

Comment:
Nit: we're creating a new function here (because of the `language`), so to make it less confusing with normal event handlers we should name this something else.

```suggestion
    const createChangeHandler = useCallback(
```